encoding/mkv.md

```diff
@@ -0,0 +1,505 @@
+
+
+## Encoding the audio
+
+We will be using FFmpeg's *libopus* and *flac* encoders.
+The following examples simply loop[^2] the command for every `ep*.wav` file,
+and all four op/ed files.
+
+For the main episodes, we will be using opus:
+
+```sh
+$ fd -e wav --regex "(_cut)" -E 'op*' -E 'ed*' -E 'nc*' -x ffmpeg -i {} -c:a libopus -b:a 128k -cutoff 20000 {.}.opus
+```
+
+And for the OP/ED's, we will use FLAC:
+
+```sh
+$ fd -e wav -E 'ep*' -x ffmpeg -i {} -c:a flac -compression_level 12 {.}.flac
+```
+
+This results in the following files:
+
+- `ep1_cut.opus` -- `ep13_cut.opus` (total: 259.7 MiB) 91.3% reduction
+- `(ed|op|nced|ncop)_cut.flac` (total: 39.8 MiB) 39.3% reduction
+
+Normally, using two different audio codecs within the same file
+would not be possible.
+However, ordered chapters make it possible for each file to use its own codec
+(as long as the track order is the same).
+
+*For the audiophiles who like to keep their anime music lossless,
+we've encoded only the OP/EDs with FLAC.
+This leaves the episode to be compressed more efficiently with a lossy codec
+such as opus, for those who don't like 'wasting' space with lossless audio.
+Since we are encoding around 90% <!-- TODO: recalculate --> of the show with opus,
+and only have the OP/ED + NCOP/NCED files occurring once,
+the filesize difference and lossless music should satisfy both groups.
+Normal dialogue should be nearly transparent at 128 kbps with opus,
+but this comes with some caveats.
+If there are 'insert' songs within an episode that you'd like to keep lossless,
+you will need to encode the entire episode with a lossless codec.*
+
+
+## Working with ordered chapters
+
+*At this point, you should have encoded clips of each episode
+plus the OP/ED/NCOP/NCED encodes.
+The following assumes these clips are named in the same `ep##.264` format
+as the .m2ts files.
+In our working directory, we should also have subtitle tracks, `ep##.ass`,
+and the encoded audio tracks `ep##_cut.opus`.
+If you used the above audio-cutting commands,
+chapter files in plaintext should also be in the folder.*
+
+At this point, you can delete the .wav files and move the .m2ts files
+out into their own folder.
+Your working directory should now look like this:
+
+![Clean Working Directory](images/mkv/example_directory.png)
+
+
+### Muxing
+
+Before we can start using ordered chapters,
+we will have to mux the current files into mkv's.
+This is because ordered chapters reference other mkv's by their UID
+to generate the virtual timeline for the player.
+MKVToolNix can save chapters *into* existing mkv files,
+so this step will be the **only time** we mux.
+
+This can be done with the MKVToolNix GUI,
+in which case the order of the tracks in the bottom list is important.
+Although mpv will only warn that tracks are mismatched,
+it is a good idea to keep the same track order in all of the mkv's.
```

Comment:
Pretty sure this is not how you pluralize abbreviations, but someone more knowledgeable should verify this.